examples/miscellanea/cartopy_gridspec.py

```diff
@@ -0,0 +1,100 @@
+"""
+Using Cartopy and AxesGrid toolkit
+----------------------------------
+This example demonstrates how to use cartopy `GeoAxes` with
+matplotlib's gridspec. Creates a list of axes for a figure
+with gridspec layout of pre-defined number of rows and columns
+
+Parameters:
+rows (int): Number of rows in grid
+cols (int): Number of columns in grid
+proj (cartopy.crs projection): cartopy.crs projection to be used
+
+Returns:
+axlist: List of axes for each individual figure in the grid.
+
+>>> make_grid(2,2,ccrs.PlateCarree())
+[<AxesSubplot:>, <AxesSubplot:>, <AxesSubplot:>, <AxesSubplot:>, <AxesSubplot:>]
+
+>>> make_grid(2,2,ccrs.PlateCarree())
+[<AxesSubplot:>, <AxesSubplot:>, <AxesSubplot:>, <AxesSubplot:>]
+"""
+
+import matplotlib.pyplot as plt
+import cartopy.crs as ccrs
+import matplotlib.gridspec as gs
+from cartopy.mpl.ticker import LongitudeFormatter, LatitudeFormatter
+import numpy as np
+
+
+def sample_data_3d(shape):
+    """Return `lons`, `lats`, `times` and fake `data`"""
+    ntimes, nlats, nlons = shape
+    lats = np.linspace(-np.pi / 2, np.pi / 2, nlats)
+    lons = np.linspace(0, 2 * np.pi, nlons)
+    lons, lats = np.meshgrid(lons, lats)
+    wave = 0.75 * (np.sin(2 * lats) ** 8) * np.cos(4 * lons)
+    mean = 0.5 * np.cos(2 * lats) * ((np.sin(2 * lats)) ** 2 + 2)
+
+    lats = np.rad2deg(lats)
+    lons = np.rad2deg(lons)
+    data = wave + mean
+
+    times = np.linspace(-1, 1, ntimes)
+    new_shape = data.shape + (ntimes,)
+    data = np.rollaxis(data.repeat(ntimes).reshape(new_shape), -1)
+    data *= times[:, np.newaxis, np.newaxis]
+
+    return lons, lats, times, data
+
+
+def make_grid(
+    rows: int = 0, cols: int = 0, proj=ccrs.PlateCarree()
+) -> list:
+
+    axlist = []
+    fig2 = plt.figure(figsize=(20, 10))
+
+    spec2 = gs.GridSpec(ncols=cols, nrows=rows, figure=fig2)
+    spec2.update(wspace=0.2, hspace=0.5)
+
+    for x in range(rows):
+        for y in range(cols):
+            axlist.append(
+                fig2.add_subplot(spec2[x, y], projection=proj)
+            )
+
+    return axlist
+
+
+def main():
+    projection = ccrs.PlateCarree()
+
+    lons, lats, times, data = sample_data_3d((6, 73, 145))
+
+    axlist = make_grid(rows=3, cols=2, proj=projection)
```

Comment:
I think we can delete the make_grid function here. I also don't think this is currently highlighting the benefits of using gridspec, it really shines when you start including slicing, so give us an example of that.

```python
fig = plt.figure()
spec = gs.GridSpec(nrows=3, ncols=3, figure=fig)
spec.update(wspace=0.2, hspace=0.5)

axlist = []
# A 1x3 plot across the entire first row
axlist.append(fig.add_subplot(spec[0, :], projection=proj))

# A 1x1 plot in the bottom left
axlist.append(fig.add_subplot(spec[0, 2], projection=proj))

# A 2x2 plot in the bottom right
axlist.append(fig.add_subplot(spec[1:, 1:], projection=proj))
```